addons/main.go

```diff
@@ -255,4 +259,32 @@ func enableClusterBootstrapAndConfigControllers(ctx context.Context, mgr ctrl.Ma
 		setupLog.Error(err, "unable to create controller", "controller", "clusterbootstrap")
 		os.Exit(1)
 	}
+
+	// Set up a ClusterCacheTracker to provide to PackageInstallStatus controller which requires a connection to remote clusters
+	l := ctrl.Log.WithName("remote").WithName("ClusterCacheTracker")
+	tracker, err := capiremote.NewClusterCacheTracker(mgr, capiremote.ClusterCacheTrackerOptions{
+		Log: &l,
+		ClientUncachedObjects: []client.Object{
+			&corev1.ConfigMap{},
+			&corev1.Secret{},
+			&corev1.Pod{},
+			&appsv1.Deployment{},
+			&appsv1.DaemonSet{},
+		},
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to create cluster cache tracker")
+		os.Exit(1)
+	}
```

Comment:
We additionally run a `ClusterCacheReconciler` which does some best effort deletion, but not sure if it's worth it. It only does something when we get an event while the cluster is already gone in the controller-runtime cache

Reply:
@sbueringer The ```healthCheckCluster``` goroutine which gets called upon adding the Watch also calls ```deleteClusterAccessor``` if cluster not found:
https://github.com/kubernetes-sigs/cluster-api/blob/main/controllers/remote/cluster_cache.go#L395

So you 'd say that doesn't suffice and the correct way is to run ``` ClusterCacheReconciler``` too, So the following?

```
if err := (&remote.ClusterCacheReconciler{
		Client:           mgr.GetClient(),
		Log:              ctrl.Log.WithName("remote").WithName("ClusterCacheReconciler"),
		Tracker:          tracker,
		WatchFilterValue: watchFilterValue,
	}).SetupWithManager(ctx, mgr, concurrency(clusterConcurrency)); err != nil {
		setupLog.Error(err, "unable to create controller", "controller", "ClusterCacheReconciler")
		os.Exit(1)
	}

```

Reply:
Did some more testing. Yup I think the reconciler should be used too. The health check in the tracker would remove the accessor at some point but the `ClusterCacheReconciler` does it more cleanly directly on Cluster delete.

Reply:
Thanks Stefan. Added ```ClusterCacheReconciler``` as well